hooks/hookexecution/executor.go

```diff
@@ -93,12 +96,12 @@ func (e *hookExecutor) ExecuteEntrypointStage(req *http.Request, body []byte) ([
 	}
 
 	payload := hookstage.EntrypointPayload{Request: req, Body: body}
-	stageOutcome, payload, stageModuleContexts, rejectErr := executeStage(executionCtx, plan, payload, handler)
-	stageOutcome.Entity = entityHttpRequest
-	stageOutcome.Stage = stageName
+	outcome, payload, contexts, rejectErr := executeStage(executionCtx, plan, payload, handler, e.metricEngine)
```

Comment:
We considered changing those functions to receiver methods but because they are generic functions they complicate the creation of `hookExecutor` type which also should use generics in its signature respectively, so we decided that it does not make sense and would bring more complexity than simplicity